pkg/volume/cinder/attacher.go

```diff
@@ -185,6 +186,14 @@ func (attacher *cinderDiskAttacher) VolumesAreAttached(specs []*volume.Spec, nod
 	}
 
 	instanceID, err := attacher.nodeInstanceID(nodeName)
+	// If node does not exist, node's volumes will be detached automatically and become available.
+	// So mark them detached.
+	if err == cloudprovider.InstanceNotFound {
+		for _, spec := range specs {
+			volumesAttachedCheck[spec] = false
```

Comment:
Checking the instance is alive is not sufficient: the cinder volume could be moved already to other nodes or in a different state. 

We need both attachment info and node info to ensure that the cinder volume is not moved elsewhere and the node the volume is attached is dead, then we can safely detach the volume.